android-o/AndroidPlayLocation/LocationUpdatesForegroundService/LocationUpdatesForegroundService/LocationUpdatesService.cs

```diff
@@ -0,0 +1,349 @@
+		}
+
+		public override void OnDestroy()
+		{
+			ServiceHandler.RemoveCallbacksAndMessages(null);
+		}
+
+		/**
+	     * Makes a request for location updates. Note that in this sample we merely log the
+	     * {@link SecurityException}.
+	     */
+		public void RequestLocationUpdates()
+		{
+			Log.Info(Tag, "Requesting location updates");
+			Utils.SetRequestingLocationUpdates(this, true);
+			StartService(new Intent(ApplicationContext, typeof(LocationUpdatesService)));
+	        try {
+	            FusedLocationClient.RequestLocationUpdates(LocationRequest, LocationCallback, Looper.MyLooper());
+	        } catch (SecurityException unlikely) {
+	            Utils.SetRequestingLocationUpdates(this, false);
+				Log.Error(Tag, "Lost location permission. Could not request updates. " + unlikely);
+			}
+		}
+
+		/**
+	     * Removes location updates. Note that in this sample we merely log the
+	     * {@link SecurityException}.
+	     */
+		public void RemoveLocationUpdates()
+		{
+			Log.Info(Tag, "Removing location updates");
+			try
+			{
+				FusedLocationClient.RemoveLocationUpdates(LocationCallback);
+				Utils.SetRequestingLocationUpdates(this, false);
+				StopSelf();
+			}
+			catch (SecurityException unlikely)
+			{
+				Utils.SetRequestingLocationUpdates(this, true);
+				Log.Error(Tag, "Lost location permission. Could not remove updates. " + unlikely);
+			}
+		}
+
+		/**
+	     * Returns the {@link NotificationCompat} used as part of the foreground service.
+	     */
+		Notification GetNotification()
+		{
+			Intent intent = new Intent(this, typeof(LocationUpdatesService));
+
+	        var text = Utils.GetLocationText(Location);
+
+			// Extra to help us figure out if we arrived in onStartCommand via the notification or not.
+			intent.PutExtra(ExtraStartedFromNotification, true);
+
+	        // The PendingIntent that leads to a call to onStartCommand() in this service.
+			var servicePendingIntent = PendingIntent.GetService(this, 0, intent, PendingIntentFlags.UpdateCurrent);
+
+			// The PendingIntent to launch activity.
+			var activityPendingIntent = PendingIntent.GetActivity(this, 0, new Intent(this, typeof(MainActivity)), 0);
+
+	        return new NotificationCompat.Builder(this)
+	                .AddAction(Resource.Drawable.ic_launch, GetString(Resource.String.launch_activity),
+	                        activityPendingIntent)
+	                .AddAction(Resource.Drawable.ic_cancel, GetString(Resource.String.remove_location_updates),
+	                        servicePendingIntent)
+	                .SetContentText(text)
+	                .SetContentTitle(Utils.GetLocationTitle(this))
+	                .SetOngoing(true)
+				    .SetPriority((int) NotificationPriority.High)
+	                .SetSmallIcon(Resource.Mipmap.ic_launcher)
+	                .SetTicker(text)
+				    .SetWhen(JavaSystem.CurrentTimeMillis())
+				    .Build();
+		}
+
+		async void GetLastLocationAsync()
```

Comment:
use `async Task` in the signature here, rather than `async void`.

Reply:
Done